src/ipc/StoreMap.cc

```diff
@@ -662,6 +671,106 @@ Ipc::StoreMap::validSlice(const int pos) const
     return 0 <= pos && pos < sliceLimit();
 }
 
+/// Checks whether the object lifetime has exceeded the specified maximum.
+/// The lifetime is considered to exceed the maximum if the time goes backwards.
+/// Uses the highest precision provided by the C++ implementation.
+class ConservativeTimer
+{
+    public:
+        typedef std::chrono::high_resolution_clock Clock;
+
+        explicit ConservativeTimer(const Clock::duration &max) :
```

Comment:
Can we justify using a (more complex) reference here as opposed to simply copying the max parameter value?